sw/device/lib/drivers/i2c/i2c.c

```diff
@@ -1,102 +1,120 @@
+**                                                                         **
+**                                                                         **
+*****************************************************************************
+	 
+*/
+
+/***************************************************************************/
+/***************************************************************************/
+
+/**
+* @file   i2c.c
+* @date   17/04/2023
+* @brief  This is the main file for the HAL of the I2C peripheral
+*
+* In this files there are definitions of low level HAL functions to interact
+* with the registers of the I2C peripheral.
+* The functionalities implemented allow to configure the peripheral with
+* parameters for the FMT and RX fifos, enable or disable interrupts and write
+* and read bytes.
+*/
+
+
+/****************************************************************************/
+/**                                                                        **/
+/*                             MODULES USED                                 */
+/**                                                                        **/
+/****************************************************************************/
 
 #include "i2c.h"
 
+#include "i2c_regs.h"
+#include "i2c_structs.h"
+
 #include "bitfield.h"
-#include "i2c_regs.h"  // Generated
+
+/****************************************************************************/
+/**                                                                        **/
+/*                        DEFINITIONS AND MACROS                            */
+/**                                                                        **/
+/****************************************************************************/
+
+static const uint32_t kNanosPerKBaud = 1000000;  // One million.
+
+/****************************************************************************/
+/**                                                                        **/
+/*                        TYPEDEFS AND STRUCTURES                           */
+/**                                                                        **/
+/****************************************************************************/
+
+/****************************************************************************/
+/**                                                                        **/
+/*                      PROTOTYPES OF LOCAL FUNCTIONS                       */
+/**                                                                        **/
+/****************************************************************************/
 
 /**
  * Performs a 32-bit integer unsigned division, rounding up. The bottom
  * 16 bits of the result are then returned.
- *
- * As usual, a divisor of 0 is still Undefined Behavior.
  */
-static uint16_t round_up_divide(uint32_t a, uint32_t b) {
-  if (a == 0) {
-    return 0;
-  }
+static uint16_t round_up_divide(uint32_t a, uint32_t b);
```

Comment:
this function should be placed into base folder, we need to create a dsp (cmsis-dsp like file(s)), let's talk about this next week